Automattic/jetpack
# Dashboard: Add backup undo feature

## Proposed changes:
* Add new `jetpack/v4/site/backup/undo-event` API on backup package to fetch the last rewindable event from the Activity Log.
* Display the last rewindable event on the `At a Glance` section in Jetpack Dashboard with a `Undo` button that will redirect the user to the restore page passing the `rewindId` of the previous rewindable event.
* Add 2 Tracks events:
  * **jetpack_wpa_aag_backup_undo_view**: when the undo feature is displayed.
  * **jetpack_wpa_aag_backup_undo_button_click**: when the user clicks on `Undo` button.

Some important notes about the feature:
* It is available for self-hosted and atomic sites with server credentials configured.
* We are looking for a valid rewindable activity in the last 20 items in the Activity Log, in case there's none, the undo feature card will not appear.

<img width="1049" alt="CleanShot 2023-08-11 at 23 24 47@2x" src="https://github.com/Automattic/jetpack/assets/1488641/b2753d03-419f-4351-8da8-97e38dfe8cec">


### Other information:

## Jetpack product discussion


* Design: p1HpG7-o15-p2
* Project: peaFOp-1Im-p2

## Does this pull request change what data or activity we track or use?


* We are tracking clicks on the new undo button.

## Testing instructions:





* Using an existing or a new Jurassic Ninja site (or Atomic), checkout to this branch using [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
* Ensure you have a VaultPress Backup plan and server credentials configured.
* Navigate to Jetpack > Dashboard and after loading the page you should see the undo section under `VaultPress Backup` card. If not, you could try uploading an image or updating/deactivating a plugin.
* Ensure that clicking on `Undo` it navigates to `https://cloud.jetpack.com/backup/{SITE_SLUG}/restore/{REWIND_ID}`.
